storage/remote/azuread/azuread_test.go

```diff
@@ -248,155 +205,90 @@ func TestTokenProvider(t *testing.T) {
-	var azureAdConfig *AzureADConfig
-	var actualTokenProvider *tokenProvider
-	var actualErr error
-
-	managedIdentityConfig := &ManagedIdentityConfig{
-		ClientID: dummyClientID,
-	}
-
-	azureAdConfig = &AzureADConfig{
-		Cloud:           "AzurePublic",
-		ManagedIdentity: managedIdentityConfig,
-	}
-	s.mockCredential.On("GetToken", mock.Anything, mock.Anything).Return(getToken(), nil)
-
-	actualTokenProvider, actualErr = newTokenProvider(azureAdConfig, s.mockCredential)
-
-	s.Assert().NotNil(actualTokenProvider)
-	s.Assert().Nil(actualErr)
-	s.Assert().NotNil(actualTokenProvider.getAccessToken(context.Background()))
-
-	oAuthConfig := &OAuthConfig{
-		ClientID:     dummyClientID,
-		ClientSecret: dummyClientSecret,
-		TenantID:     dummyTenantID,
-	}
-
-	azureAdConfig = &AzureADConfig{
-		Cloud: "AzurePublic",
-		OAuth: oAuthConfig,
-	}
-
-	actualTokenProvider, actualErr = newTokenProvider(azureAdConfig, s.mockCredential)
-
-	s.Assert().NotNil(actualTokenProvider)
-	s.Assert().Nil(actualErr)
-	s.Assert().NotNil(actualTokenProvider.getAccessToken(context.Background()))
-}
-
-func (s *TokenProviderTestSuite) TestPeriodicTokenRefresh_ManagedIdentity_Success() {
-	// setup
-	managedIdentityConfig := &ManagedIdentityConfig{
-		ClientID: dummyClientID,
-	}
-
-	azureAdConfig := &AzureADConfig{
-		Cloud:           "AzurePublic",
-		ManagedIdentity: managedIdentityConfig,
-	}
-	testToken := &azcore.AccessToken{
-		Token:     testTokenString,
-		ExpiresOn: testTokenExpiry,
+func (s *TokenProviderTestSuite) TestNewTokenProvider() {
+	cases := []struct {
+		cfg *AzureADConfig
+		err string
+	}{
+		// Invalid tokenProvider for managedidentity.
+		{
+			cfg: &AzureADConfig{
+				Cloud: "PublicAzure",
+				ManagedIdentity: &ManagedIdentityConfig{
+					ClientID: dummyClientID,
+				},
+			},
+			err: "Cloud is not specified or is incorrect: ",
+		},
+		// Invalid tokenProvider for oauth.
+		{
+			cfg: &AzureADConfig{
+				Cloud: "PublicAzure",
+				OAuth: &OAuthConfig{
+					ClientID:     dummyClientID,
+					ClientSecret: dummyClientSecret,
+					TenantID:     dummyTenantID,
+				},
+			},
+			err: "Cloud is not specified or is incorrect: ",
+		},
+		// Valid tokenProvider for managedidentity.
+		{
+			cfg: &AzureADConfig{
+				Cloud: "AzurePublic",
+				ManagedIdentity: &ManagedIdentityConfig{
+					ClientID: dummyClientID,
+				},
+			},
+		},
+		// Valid tokenProvider for oauth.
+		{
+			cfg: &AzureADConfig{
+				Cloud: "AzurePublic",
+				OAuth: &OAuthConfig{
+					ClientID:     dummyClientID,
+					ClientSecret: dummyClientSecret,
+					TenantID:     dummyTenantID,
+				},
+			},
+		},
+	}
+	mockGetTokenCallCounter := 1
+	for _, c := range cases {
+		if c.err != "" {
+			actualTokenProvider, actualErr := newTokenProvider(c.cfg, s.mockCredential)
+
+			s.Assert().Nil(actualTokenProvider)
+			s.Assert().NotNil(actualErr)
+			s.Assert().Equal("Cloud is not specified or is incorrect: "+c.cfg.Cloud, actualErr.Error())
```

Comment:
I think it might be cleaner here to use [ErrorContains](https://pkg.go.dev/github.com/stretchr/testify@v1.8.4/assert#Assertions.ErrorContains) with `c.err` instead of duplicating it here.

Reply:
Replaced the assert condition with `ErrorContains`